File: lab4/lang-tests/test-2/typed.java
public class typed {
    public static void main(String[] args) {

        int num = 10; 
        System.out.println();
        System.out.println("la variable num:" + num);

        String word = new String("hola!!");
        System.out.println("la variable word: " + word);

        String word2 = new String("chau!!");
        System.out.println("la variable word2: " + word2);
        System.out.println();

        String result = word + num;
        System.out.println("result = word + num");
        System.out.println("result: " + result);
        System.out.println();

        System.out.println("word * num es error");
        System.out.println("word * word es error");
        System.out.println("word + num es valido : " + result);
        System.out.println("word + word es valido: " + (word + word2));
        System.out.println();
        System.out.println("java es de tipado DEBIL");

    }
}

File: lab4/lang-tests/test-2/typed.scala
object Typed {
   def main(args: Array[String]) {
      println("prueba de tipado fuerte o debil")
      println()
      
      var num : Int = 10
      println("variable num: ")
      println(num)
      println()
      
      var word : String = "hola!!"
      println("variable word: ")
      println(word)
      println()
      
      var result = word * num
      println(result)
      println()
      println("concluimos que scala es de tipado DEBIL")
   }
}